src/codegen/bindgen.ts

```diff
@@ -0,0 +1,1262 @@
+        cpp.line(
+          `${storageLocation} = Bun::toString(WebCore::convert<WebCore::IDLDOMString>(*global, ${jsValueRef}));`,
+        );
+        cpp.line(`RETURN_IF_EXCEPTION(throwScope, {});`);
+        break;
+      case "dictionary": {
+        if (decl === "declare") cpp.line(`${storageLocation};`);
+        cpp.line(`if (!convert${type.cppInternalName()}(&${storageLocation}, global, ${jsValueRef}))`);
+        cpp.indent();
+        cpp.line(`return {};`);
+        cpp.dedent();
+        break;
+      }
+      default:
+        throw new Error(`TODO: emitConvertValue for Type ${type.kind}`);
+    }
+  }
+}
+
+/**
+ * The built in WebCore range adaptors do not support arbitrary ranges, but that
+ * is something we want to have. They aren't common, so they are just tacked
+ * onto the webkit one.
+ */
+function emitRangeModifierCheck(
+  cAbiType: CAbiType,
+  storageLocation: string,
+  range: ["clamp" | "enforce", bigint, bigint],
+) {
+  const [kind, min, max] = range;
+  if (kind === "clamp") {
+    cpp.line(`if (${storageLocation} < ${min}) ${storageLocation} = ${min};`);
+    cpp.line(`else if (${storageLocation} > ${max}) ${storageLocation} = ${max};`);
+  } else if (kind === "enforce") {
+    cpp.line(`if (${storageLocation} < ${min} || ${storageLocation} > ${max}) {`);
+    cpp.indent();
+    cpp.line(
+      `throwTypeError(global, throwScope, rangeErrorString<${cAbiTypeName(cAbiType)}>(${storageLocation}, ${min}, ${max}));`,
+    );
+    cpp.line(`return {};`);
+    cpp.dedent();
+    cpp.line(`}`);
+  } else {
+    throw new Error(`TODO: range modifier ${kind}`);
+  }
+}
+
+function addHeaderForType(type: TypeImpl) {
+  if (type.lowersToNamedType() && type.ownerFile) {
+    headers.add(`Generated${cap(type.ownerFile)}.h`);
+  }
+}
+
+function emitConvertDictionaryFunction(type: TypeImpl) {
+  assert(type.kind === "dictionary");
+  const fields = type.data as DictionaryField[];
+
+  addHeaderForType(type);
+
+  cpp.line(`// Internal dictionary parse for ${type.name()}`);
+  cpp.line(
+    `bool convert${type.cppInternalName()}(${type.cppName()}* result, JSC::JSGlobalObject* global, JSC::JSValue value) {`,
+  );
+  cpp.indent();
+
+  cpp.line(`auto& vm = JSC::getVM(global);`);
+  cpp.line(`auto throwScope = DECLARE_THROW_SCOPE(vm);`);
+  cpp.line(`bool isNullOrUndefined = value.isUndefinedOrNull();`);
+  cpp.line(`auto* object = isNullOrUndefined ? nullptr : value.getObject();`);
+  cpp.line(`if (UNLIKELY(!isNullOrUndefined && !object)) {`);
+  cpp.line(`    throwTypeError(global, throwScope);`);
+  cpp.line(`    return false;`);
+  cpp.line(`}`);
+  cpp.line(`JSC::JSValue propValue;`);
+
+  for (const field of fields) {
+    const { key, type: fieldType } = field;
+    cpp.line("// " + key);
+    cpp.line(`if (isNullOrUndefined) {`);
+    cpp.line(`    propValue = JSC::jsUndefined();`);
+    cpp.line(`} else {`);
+    cpp.line(`    propValue = object->get(global, JSC::Identifier::fromString(vm, ${str(key)}_s));`);
```

Comment:
Use our prototype pollution mitigation
```suggestion
    cpp.line(`    propValue = Bun::getIfPropertyExistsPrototypePollutionMitigation(global, JSC::Identifier::fromString(vm, ${str(key)}_s));`);
```